modules/images/webp-uploads/load.php

```diff
@@ -96,6 +96,34 @@ function webp_uploads_create_sources_property( array $metadata, $attachment_id )
 			continue;
 		}
 
+		$original_image_filesize = isset( $metadata['filesize'] ) ? (int) $metadata['filesize'] : 0;
+		$webp_image_filesize     = isset( $image['filesize'] ) ? (int) $image['filesize'] : 0;
+		if ( $original_image_filesize <= 0 && $webp_image_filesize <= 0 ) {
+			continue;
+		}
+
+		/**
+		 * Filter whether to prioritize the file type over the mime type.
+		 *
+		 * By default the performance lab plugin will use the mime type with the smaller filesize
```

Comment:
This documentation can be improved a bit - the filter controls whether to retain WebP images even if they are larger than the jpegs, right?

Maybe something like
"Filter whether WebP images that are larger than the matching JPEG should be discarded. Default is true"

Also makes me think the filter name should reflect this as well, maybe something like `webp_uploads_discard_larger_generated_images`, curious what @felixarntz thinks since he is great at naming things.

Reply:
Thanks for the feedback. I have updated the documentation for the filter and updated the filter name to `webp_uploads_discard_larger_generated_images` for now. Will update it when Felix reviews this and suggests the changes regarding the filter name.